dtfs-central-api/api-tests/v1/bank/banks-get.api-test.ts

```diff
@@ -0,0 +1,102 @@
+import { Bank, MonthAndYear, UtilisationReportEntity, UtilisationReportEntityMockBuilder, UtilisationReportReconciliationStatus } from '@ukef/dtfs2-common';
+import wipeDB from '../../wipeDB';
+import { testApi } from '../../test-api';
+import { SqlDbHelper } from '../../sql-db-helper';
+import { aBank, getSqlIdGenerator } from '../../../test-helpers';
+import { mongoDbClient } from '../../../src/drivers/db-client';
+
+const BASE_URL = '/v1/bank';
+
+describe(`GET ${BASE_URL}`, () => {
+  const BANK_ID_ONE = '123';
+  const BANK_ID_TWO = '456';
+  const BANK_ID_THREE = '789';
+  const MOCK_BANKS: Bank[] = [
+    { ...aBank(), id: BANK_ID_ONE, _id: new ObjectId(), isVisibleInTfmUtilisationReports: true },
+    { ...aBank(), id: BANK_ID_TWO, _id: new ObjectId(), isVisibleInTfmUtilisationReports: true },
+    { ...aBank(), id: BANK_ID_THREE, _id: new ObjectId(), isVisibleInTfmUtilisationReports: false },
+  ];
+
+  beforeEach(async () => {
+    await wipeDB.wipe(['banks']);
+    await SqlDbHelper.deleteAllEntries('UtilisationReport');
+
+    const banksCollection = await mongoDbClient.getCollection('banks');
+    await banksCollection.insertMany(MOCK_BANKS);
+  });
+
+  afterAll(async () => {
+    await wipeDB.wipe(['banks']);
+    await SqlDbHelper.deleteAllEntries('UtilisationReport');
+  });
+
+  describe.each([false, undefined])("when the 'includeReportingYears' query is set to '%s'", (includeReportingYears) => {
+    it('responds with a 200 (Ok) and all banks without reporting years', async () => {
+      // Arrange
+      const requestUrl = axios.getUri({
+        url: BASE_URL,
+        params: { includeReportingYears },
+      });
+
+      const banks = MOCK_BANKS.map((bank) => ({ ...bank, _id: bank._id.toString() }));
+
+      // Act
+      const response = await testApi.get(requestUrl);
+
+      // Assert
+      expect(response.status).toBe(axios.HttpStatusCode.Ok);
+      expect(response.body).toEqual(banks);
+    });
+  });
+
+  describe("when the 'includeReportingYears' query is set to 'true'", () => {
+    const reportIdGenerator = getSqlIdGenerator();
+
+    const aUtilisationReportForStatusBankIdAndMonthlyReportPeriod = (
+      status: UtilisationReportReconciliationStatus,
+      bankId: string,
+      monthAndYear: MonthAndYear,
+    ): UtilisationReportEntity =>
+      UtilisationReportEntityMockBuilder.forStatus(status)
+        .withId(reportIdGenerator.next().value)
+        .withBankId(bankId)
+        .withReportPeriod({
+          start: monthAndYear,
+          end: monthAndYear,
+        })
+        .build();
+
+    it('responds with a 200 (Ok) and all banks with reporting years', async () => {
+      // Arrange
+      const requestUrl = axios.getUri({
+        url: BASE_URL,
+        params: { includeReportingYears: true },
+      });
+
+      const utilisationReports = [
+        aUtilisationReportForStatusBankIdAndMonthlyReportPeriod('PENDING_RECONCILIATION', BANK_ID_ONE, { month: 1, year: 2021 }),
```

Comment:
* Variable names are too long.
* CAn these values please be picked from constants?

Reply:
Could you please make a suggestion for another name which retains the same context? If we were to inline this using our builders, it would become

```typescript
const utilisationReports = [
  UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION')
    .withBankId(BANK_ID_ONE)
    .withReportPeriod({
      start: { month: 1, year: 2021 },
      end: { month: 1, year: 2021 },
    })
    .build(),
    // ... repeated for each case
];
```

which I'm sure you can agree is much more bloated

Reply:
Can it be rather `monthlyUtilisationBankReport` or `monthlyUtlisationBankOneReport`?

Reply:
I have replaced it with the less clear but shorter `aUtilisationReportFor` helper - hopefully this is now more readable